packages/shared/lib/services/proxy.service.ts

```diff
@@ -77,52 +77,39 @@ class ProxyService {
         let token;
         switch (connection.credentials.type) {
             case 'OAUTH2':
-                {
-                    const credentials = connection.credentials;
-                    token = credentials.access_token;
-                }
+            case 'APP': {
+                const credentials = connection.credentials;
+                token = credentials.access_token;
                 break;
-            case 'OAUTH1': {
-                const error = new Error('OAuth1 is not supported yet in the proxy.');
-                const nangoError = new NangoError('pass_through_error', error);
-                return { success: false, error: nangoError, response: null, logs };
             }
-            case 'BASIC':
-                token = connection.credentials;
-                break;
-            case 'API_KEY':
-                token = connection.credentials;
-                break;
-            case 'APP':
-                {
-                    const credentials = connection.credentials;
-                    token = credentials.access_token;
-                }
-                break;
             case 'OAUTH2_CC':
-                {
-                    const credentials = connection.credentials;
-                    token = credentials.token;
-                }
-                break;
+            case 'TWO_STEP':
             case 'TABLEAU':
-                {
-                    const credentials = connection.credentials;
-                    token = credentials.token;
-                }
-                break;
             case 'JWT':
-                {
-                    const credentials = connection.credentials;
-                    token = credentials.token;
-                }
+            case 'SIGNATURE': {
+                const credentials = connection.credentials;
+                token = credentials.token;
                 break;
-            case 'SIGNATURE':
-                {
-                    const credentials = connection.credentials;
-                    token = credentials.token;
-                }
+            }
+            case 'BASIC':
+            case 'API_KEY':
+                token = connection.credentials;
                 break;
+            case 'OAUTH1': {
+                const error = new Error('OAuth1 is not supported yet in the proxy.');
+                const nangoError = new NangoError('pass_through_error', error);
+                return { success: false, error: nangoError, response: null, logs };
+            }
+            case 'APP_STORE':
+            case 'CUSTOM':
+            case 'TBA':
+            case undefined:
+            case 'BILL': {
+                break;
+            }
+            default: {
+                throw new Error(`Unhandled connection.credentials for: ${connection.credentials}`);
```

Comment:
Passing this could work but would also leak credentials, typescript is complaining about never but I think it's acceptable to type it as any in that case `((connection.credentials as any).type`